src/rewrites.rs

```diff
@@ -849,27 +1037,519 @@ impl MultiPatterns {
                 if compatible(&subst_1_dec, &subst_2_dec, &map_1.var_map) {
                     // If so, merge two substitutions
                     let merged_subst = merge_subst(subst_1_dec, subst_2_dec, &map_1.var_map);
+                    // Check if any source pattern contains blacklisted nodes
+                    if self.filter_after {
+                        if contains_blacklist(
```

Comment:
This condition is very long. Define a temp variable for this.

Reply:
Done